tracer/src/Datadog.Trace/Debugger/Snapshots/DebuggerSnapshotCreator.cs

```diff
@@ -313,7 +313,8 @@ internal DebuggerSnapshotCreator AddGeneralInfo(string service, string traceId,
 
         public DebuggerSnapshotCreator AddMessage()
         {
-            var snapshotObject = JsonConvert.DeserializeObject<Snapshot>(_jsonUnderlyingString.ToString() + "}");
+            _jsonUnderlyingString.Append('}');
+            var snapshotObject = JsonConvert.DeserializeObject<Snapshot>(_jsonUnderlyingString.ToString());
```

Comment:
Note: this is a behaviour change, which may not be correct. For example, if `AddMessage()` is called multiple times in quick succession, you will have `}` appended repeatedly at the end. One option you could take is to remove the added char after adding.
```suggestion
            _jsonUnderlyingString.Append('}');
            var snapshotObject = JsonConvert.DeserializeObject<Snapshot>(_jsonUnderlyingString.ToString());
            _jsonUnderlyingString.Remove(_jsonUnderlyingString.Length - 1, 1);
```



Also, unrelated, line 22 I think can be removed as it's unused
`private static readonly ImmutableDebuggerSettings Settings = ImmutableDebuggerSettings.Create(DebuggerSettings.FromDefaultSource());`

Can't comment on it directly as you didn't change that line 🙂

Reply:
You are absolutely right.
It was fixed here dbc7c6061e7997f02ebca966f53795bf1e5181a4

It doesn't looks nice, but I think it still better, than allocating two pretty big strings, as it was before.